### PR TITLE
Various fixes from RN upgrade work

### PR DESCRIPTION
**Pull request by pedropombeiro**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

### Summary
This PR includes some work which doesn't need to be part of the RN upgrade PR. Namely, it:
- improves the way that the `versionCode` is passed to the Gradle script, but passing it as a command line argument instead of patching the `android/gradle.properties` file;
- Ensures that `nix/mobile/android/default.nix` defaults to `android` as `target-os`;
- Removes orphaned `scripts/release-android.sh` script;
- Centralizes Gradle script version requirements in `android/gradle.properties`;
- Upgrades `react-native-fs` to a version that is compatible with the RN version we're using;
- Ensures that node_modules is reset to the correct version when yarn.lock changes (e.g. when changing branches).

Upgraded `minSdkVersion` due to the following error:

```
> Task :app:processReleaseManifest FAILED
/build/status-react-source-release-android/android/app/src/release/AndroidManifest.xml Error:
        uses-sdk:minSdkVersion 16 cannot be smaller than version 18 declared in library [:react-native-status-keycard] /build/status-react-source-release-android/node_modules/react-native-status-keycard/android/build/intermediates/library_manifest/release/AndroidManifest.xml as the library might be using APIs not available in 16
        Suggestion: use a compatible library with a minSdk of at most 16,
                or increase this project's minSdk version to at least 18,
                or use tools:overrideLibrary="com.reactlibrary" to force usage (may lead to runtime failures)
```

status: ready

### CODE REVIEW

File: nix/mobile/android/targets/release-android.nix [hunk 1]
```diff
@@ -113,7 +113,7 @@ in stdenv.mkDerivation {
     ${concatStrings (catAttrs "shellHook" [ mavenAndNpmDeps status-go ])}
 
     pushd android
-    ${adhocEnvVars} gradle assemble${capitalizedBuildType} -S -Dmaven.repo.local='${localMavenRepo}' --offline ${gradle-opts} || exit
+    ${adhocEnvVars} ./gradlew -PversionCode=${build-number} assemble${capitalizedBuildType} || exit
```
Comment (jakubgs) [inline nix/mobile/android/targets/release-android.nix:116]: Wait, so when is it set to the actual value of the `versionNumber`, not the `9999`?

Comment (pedropombeiro) [reply nix/mobile/android/targets/release-android.nix:116]: Right, the command line arg overrides `gradle.properties`

Comment (jakubgs) [reply nix/mobile/android/targets/release-android.nix:116]: Right, but the `versionNumber` has to be set based on the result of `scripts/build_no.sh` in CI. How does that happen?

Comment (pedropombeiro) [reply nix/mobile/android/targets/release-android.nix:116]: That's happening here: https://github.com/status-im/status-react/blob/fix%2Frn-upgrade-misc/ci/utils.groovy#L66-L72

Comment (jakubgs) [reply nix/mobile/android/targets/release-android.nix:116]: Ah yes, i was confused by the method in the Gradle file.

**Review by jakubgs — APPROVED** [review]
Nice.

**Review by flexsurfer — APPROVED** [review]

**Review by yenda — APPROVED** [review]

**Comment by pedropombeiro** [comment]
@churik are these e2e test failures expected? I didn't change anything in the software in this PR.

**Comment by churik** [comment]
@PombeirP yes, we changed workflow in `develop`. Triggered tests from branch - let's wait for results.

**Comment by pedropombeiro** [comment]
@churik what do these results mean? Is it OK to merge?

**Comment by churik** [comment]
@PombeirP failed e2e are not related to your PR.
Ready to merge.

**Pull request merged**